ansible/roles/baselayout/tasks/ccache.yml

```diff
@@ -4,13 +4,13 @@
 # @requires: version=3.2.4
 #
 
-- name: "ccache : latest ccache release version"
-  local_action: shell curl -sL https://www.samba.org/ftp/ccache/ | awk '{ if (match($0, />ccache-(([0-9]+)\.([0-9]+)\.?([0-9]+)?)\.tar\.gz</, a)) { if (a[2]+0 > maxmaj || (a[2]+0 == maxmaj && a[3]+0 > maxmin) || (a[2]+0 == maxmaj && a[3]+0 == maxmin && a[4]+0 > maxpat)) { maxmaj = a[2]+0; maxmin = a[3]+0; maxpat = a[4]+0; } } } END { printf "%s.%s.%s\n", maxmaj, maxmin, maxpat }'
-  register: ccache_latest_result
+- name: "ccache : fetch samba ccache HTML page"
+  local_action: shell curl -sL https://www.samba.org/ftp/ccache/
+  register: ccache_html_content
 
-- name: "ccache : extract version string"
+- name: "ccache : extract ccache latest version"
   set_fact:
-    ccache_latest: "{{ ccache_latest_result.stdout }}"
+    ccache_latest: "{{ ccache_html_content.stdout | regex_findall('ccache-[0-9]+.[0-9]+(?:.[0-9]+)*.tar.gz') | map('regex_replace', 'ccache-') | map('regex_replace', '.tar.gz') | list | latest_version }}"
```

Comment:
Some comment went lost between rebases, as part of this I've changed the regex moving from something matching:

`x.y` and `x.y.z`

to something matching versions with 2 or more digits

`x.y`, `x.y.z`, `x.y.z.t` etc

This was mainly done because I've found files like `git-1.8.3.2.tar.gz` in the git web page (https://www.kernel.org/pub/software/scm/git/).

Do you have any opinion on this?
Should we stick to match only semver versions or should we allow for 4+ digits versions?